packages/legend-application-studio/src/stores/LegendStudioClientSideStorageUtils.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { WebApplicationClientSideStorage } from '@finos/legend-application';
+
+export enum LEGEND_STUDIO_LOCAL_STORAGE_KEY {
```

Comment:
Lets clean this up a bit. First I think what we should store is something like USER_CONFIG as key and json string as its value. 

We want to leverage our serialization here when reading from the storage and not have definitions such as `BOOLEAN_VALUES`. 
So I would define a class maybe like `SavedUserConfig`. When reading from the storage if the user config is there deseralize it to this class else use a default config. When saving to the storage interacte with the `SavedUserConfig` and then seralize to JSON -> string when saving it in the storage.

I don't think this scales as much as I would want it to at the moment.